### ISSUE TITLE
API is not compatible with requests containing spcial/encoding char

### ISSUE THREAD

**Issue by Smosia** [nicolargo/glances#2820]
**Describe the bug**
When I am trying to get info about mnt_point through API, an empty json is returned.

**To Reproduce**
1. curl http://localhost:61208/api/4/fs/mnt_point
Return:
```
{
  "mnt_point": [
    "/NAS",
    "/etc/resolv.conf",
    "/etc/hostname",
    "/etc/hosts"
  ]
}
```
2. curl http://localhost:61208/api/4/fs/mnt_point/NAS
```
Return:
{
  "NAS": []
}
```
**Expected behavior**
As written in [docs](https://glances.readthedocs.io/en/latest/api.html#get-fs):
```
# curl http://localhost:61208/api/3/fs/mnt_point//
{"/": [{"device_name": "/dev/mapper/ubuntu--gnome--vg-root",
        "free": 4763168768,
        "fs_type": "ext4",
        "key": "mnt_point",
        "mnt_point": "/",
        "percent": 97.9,
        "size": 243334156288,
        "used": 226183532544}]}
```

**Environement (please complete the following information)**
 - Operating System Debian GNU/Linux 12 (bookworm)
 - Glances version: 4.0.7
 - How do you install Glances: docker container through compose file
```
version: '3'
services:
  monitoring:
    image: nicolargo/glances:latest-full
    pid: host
    restart: always
    network_mode: host
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /run/user/1000/podman/podman.sock:/run/user/1000/podman/podman.sock
      - /media/NAS:/NAS:ro
    environment:
      - "GLANCES_OPT=-w"
    ports:
      - "61208-61209:61208-61209"
```

**Comment by nicolargo** [comment]
It works with other field:

```
glances on  develop [!] via 🐍 v3.12.3 
➜ curl -s http://localhost:61208/api/4/fs | jq .
[
  {
    "device_name": "/dev/mapper/ubuntu--vg-ubuntu--lv",
    "fs_type": "ext4",
    "mnt_point": "/",
    "size": 1003736440832,
    "used": 50393305088,
    "free": 902280630272,
    "percent": 5.3,
    "key": "mnt_point"
  }
]

glances on  develop [!] via 🐍 v3.12.3 
➜ curl -s http://localhost:61208/api/4/fs/fs_type/ext4 | jq .
{
  "ext4": [
    {
      "device_name": "/dev/mapper/ubuntu--vg-ubuntu--lv",
      "fs_type": "ext4",
      "mnt_point": "/",
      "size": 1003736440832,
      "used": 50393305088,
      "free": 902280630272,
      "percent": 5.3,
      "key": "mnt_point"
    }
  ]
}
```

But not with the mnt_point:

```
glances on  develop [!] via 🐍 v3.12.3 
➜ curl -s http://localhost:61208/api/4/fs/mnt_point | jq .
{
  "mnt_point": [
    "/"
  ]
}

glances on  develop [!] via 🐍 v3.12.3 
➜ curl -s http://localhost:61208/api/4/fs/mnt_point// | jq .
>>> do not work with /

glances on  develop [!] via 🐍 v3.12.3 
❯ curl -s http://localhost:61208/api/4/fs/mnt_point/%2F | jq .
>>> do not work with / encoded

glances on  develop [!] via 🐍 v3.12.3 
➜ curl -s http://localhost:61208/api/4/fs/mnt_point/\/ | jq .
>>> do not work with / backslashed...
```

After a quick debug, the method _api_value is not called when a special char (encoded or not is used). Have to have a look on the FastAPI side....

**Comment by nicolargo** [comment]
May be related to https://github.com/tiangolo/fastapi/issues/4390
So perhaps, the issue came from Uvicorn.

Ok find a solution here: https://github.com/tiangolo/fastapi/issues/4390#issuecomment-1019558295

If i change the Glances code from:

```
        router.add_api_route(
            f'/api/{self.API_VERSION}/{{plugin}}/{{item}}/{{value}}',
            response_class=ORJSONResponse,
            endpoint=self._api_value,
        )
```

to ```
        router.add_api_route(
            f'/api/{self.API_VERSION}/{{plugin}}/{{item}}/{{value:path}}',
            response_class=ORJSONResponse,
            endpoint=self._api_value,
        )
```

then it works as expected when i use the encoded URL:

```
curl -s http://localhost:61208/api/4/fs/mnt_point/%2F | jq .
{
  "/": [
    {
      "device_name": "/dev/mapper/ubuntu--vg-ubuntu--lv",
      "fs_type": "ext4",
      "mnt_point": "/",
      "size": 1003736440832,
      "used": 50395852800,
      "free": 902278082560,
      "percent": 5.3,
      "key": "mnt_point"
    }
  ]
}
```

Had to check the impact of this change on other use case (when the  value do not need to be encoded).

**Comment by nicolargo** [comment]
WebUI and API tests are OK.

Merged into develop.

Will be released in Glances 4.1.0.

Thanks for the issue report @Smosia

**Comment by Smosia** [comment]
Thank you. 
I'll check when the docker container is updated.
